### PR TITLE
Add costs endpoint for getting costs/savings for a managed instance

### PR DESCRIPTION
**Pull request by tenyo**

### CODE REVIEW

File: spotinst/managedinstance.go [hunk 1]
```diff
@@ -40,6 +40,32 @@ func (m *ManagedInstance) GetAWSManagedInstanceByID(ctx context.Context, id stri
...
+		ManagedInstanceID: spotinst.String(id),
+	}
+
+	if fromDate != "" && toDate != "" {
+		input.FromDate = spotinst.String(fromDate)
+		input.ToDate = spotinst.String(toDate)
+	}
+
+	output, err := m.Service.CloudProviderAWS().Costs(ctx, input)
+
```
Comment (fishnix) [inline spotinst/managedinstance.go]: ✂️

File: spotinst/managedinstance.go [hunk 2]
```diff
@@ -40,6 +40,32 @@ func (m *ManagedInstance) GetAWSManagedInstanceByID(ctx context.Context, id stri
...
+func (m *ManagedInstance) GetAWSManagedInstanceCostsByID(ctx context.Context, id, fromDate, toDate string) (*aws.CostsManagedInstanceOutput, error) {
+	if id == "" {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	log.Infof("getting costs details for aws managed instance: %s (fromDate: %s, toDate: %s)", id, fromDate, toDate)
+
+	input := &aws.CostsManagedInstanceInput{
+		ManagedInstanceID: spotinst.String(id),
+	}
```
Comment (fishnix) [inline spotinst/managedinstance.go:52]: I think we usually take the pointer in the call, not in the declaration... minor but 🤷

**Review by fishnix — APPROVED** [review]
looks good - a minor comment or two.  this looks a lot like the costexplorer endpoints, I wonder if they use that in the backend. 👍

**Comment by damnski** [comment]
:+1:

**Pull request merged**